src/protocols/secure_channel/CASESession.cpp

```diff
@@ -325,7 +326,10 @@ CHIP_ERROR CASESession::SendSigmaR1()
         bbuf.Put16(n_trusted_roots);
         for (uint16_t i = 0; i < n_trusted_roots; ++i)
         {
-            bbuf.Put(mOpCredSet->GetTrustedRootId(i)->mId, kTrustedRootIdSize);
+            if (mOpCredSet->GetTrustedRootId(i) != nullptr && mOpCredSet->GetTrustedRootId(i)->mId != nullptr)
+            {
+                bbuf.Put(mOpCredSet->GetTrustedRootId(i)->mId, kTrustedRootIdSize);
+            }
```

Comment:
Trusted roots do not belong in Sigma1 anymore. The destination ID allows immediate determination of the correct root on receipt of Sigma1.

Reply:
The message structure will change when we update the codec to use TLV. I'll leave this cleanup as part of that..